### DISCUSSION TITLE
ok i'm stumped

### DISCUSSION THREAD

**Discussion by mdefrancesco** [MMRIZE/MMM-CalendarExt3#124] [Q&A]
MMM-CalenderExt3
i feel like i tried everything before i came on here and asked for help. I followed the install (not upgrade) instructions, pretty straightforward and when i ran the magic mirror i just get a very small DECEMBER at the bottom of the screen, with a line underneath it and thats it. no calendars, no errors, no nothing. 
i have a bunch of other modules running fine, but to be sure, i went back and used a completely clean config.js file and got the same thing. i tried using the sample config file included in the packet and still nothing. i even got a fresh calendar.js in case i tweaked something at some point. 

![image](https://github.com/MMRIZE/MMM-CalendarExt3/assets/123851794/96ddca03-390b-4f2f-bbf4-03febc72f9cf)

like i said, followed the instructions exactly.

just looking for some ideas and where to start looking. or maybe someone has come across the same issue?

thanks in advance
-M

**Comment by eouia** [comment]
Show me front-dev console log(Ctrl+Shif+i maybe)

**Comment by mdefrancesco** [comment]
i assume this could be a part of the problem

![image](https://github.com/MMRIZE/MMM-CalendarExt3/assets/123851794/3e8f9d17-db55-4922-8dd6-a22cc853803c)

**Reply by eouia** [reply]
What is your MM version? And Chromium/Electron version?

**Comment by mdefrancesco** [comment]
so MM v2.22.0
Chromium Version 116.0.5845.102 (Official Build) Built on Raspbian , running on Raspbian 11 (32-bit)

and if you're going to say update either of those, then i'll need a little guidance to not mess the whole thing up please :-)

**Reply by eouia** [reply]
Chromium 116 you said may be your desktop browser, not related to MM's Electron. Your MM 2.22 is using individual Chromium 108.
`toSorted` has been introduced in JS as a convenient new feature since Chromium 110. So you may need newer electron/chromium of MM.(you need to update MM)

**Reply by mdefrancesco** [reply]
so if i update MM (whats the latest version right now) it will automatically update chromium to 110? do you think that might be my issue?

and thanks for the attention to this. i'm pretty stubborn and very frustrated i can't figure this out.

**Reply by eouia** [reply]
Latest MM 2.25 is using Electron 26 which has Chromium 116. So it will work I think.

**Reply by mdefrancesco** [reply]
awesome, i'm backing up my MM folders right now and then will try the upgrade
thanks!

**Comment by mdefrancesco** [comment]
ok, upgraded to 2.25 and this sparked a while new issue. Now MM isn't even starting and i get this error in the terminal after i run npm start

> magicmirror@2.25.0 start
> DISPLAY="${DISPLAY:=:0}" ./node_modules/.bin/electron js/electron.js

[24.12.2023 19:27.26.394] [LOG]   Starting MagicMirror: v2.25.0
[24.12.2023 19:27.26.403] [LOG]   Loading config ...
[24.12.2023 19:27.26.408] [DEBUG] config template file not exists, no envsubst
[24.12.2023 19:27.26.416] [LOG]   Loading module helpers ...
[24.12.2023 19:27.26.419] [LOG]   No helper found for module: alert.
[24.12.2023 19:27.26.473] [LOG]   Initializing new module helper ...
[24.12.2023 19:27.26.478] [LOG]   Module helper loaded: updatenotification
[24.12.2023 19:27.26.481] [LOG]   No helper found for module: clock.
[24.12.2023 19:27.26.848] [LOG]   Initializing new module helper ...
[24.12.2023 19:27.26.850] [LOG]   Module helper loaded: calendar
[24.12.2023 19:27.26.852] [LOG]   No helper found for module: MMM-CalendarExt3.
[24.12.2023 19:27.27.005] [LOG]   Initializing new module helper ...
[24.12.2023 19:27.27.006] [LOG]   Module helper loaded: MMM-OpenWeatherForecast
[24.12.2023 19:27.27.007] [LOG]   No helper found for module: MMM-page-indicator.
[24.12.2023 19:27.27.026] [ERROR] (node:3092) UnhandledPromiseRejectionWarning: Error: Cannot find module 'fetch'
Require stack:
- /home/pi/MagicMirror/modules/MMM-MyCalendar/calendarfetcher.js
- /home/pi/MagicMirror/modules/MMM-MyCalendar/node_helper.js
- /home/pi/MagicMirror/js/app.js
- /home/pi/MagicMirror/js/electron.js
- /home/pi/MagicMirror/node_modules/electron/dist/resources/default_app.asar/main.js
- 
    at node:internal/modules/cjs/loader:1084:15
    at Function.<anonymous> (node:electron/js2c/browser_init:2:117419)
    at Module._resolveFilename (/home/pi/MagicMirror/node_modules/module-alias/index.js:49:29)
    at node:internal/modules/cjs/loader:929:27
    at Function._load (node:electron/js2c/asar_bundle:2:13327)
    at Module.require (node:internal/modules/cjs/loader:1150:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-MyCalendar/calendarfetcher.js:11:15)
    at Module._compile (node:internal/modules/cjs/loader:1271:14)
    at Object..js (node:internal/modules/cjs/loader:1326:10)
    at Module.load (node:internal/modules/cjs/loader:1126:32)
    at node:internal/modules/cjs/loader:967:12
    at Function._load (node:electron/js2c/asar_bundle:2:13327)
    at Module.require (node:internal/modules/cjs/loader:1150:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-MyCalendar/node_helper.js:8:25)
(Use `electron --trace-warnings ...` to show where the warning was created)
[24.12.2023 19:27.27.027] [ERROR] (node:3092) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
[3092:1224/192727.547689:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files
[3127:1224/192729.148349:ERROR:viz_main_impl.cc(186)] Exiting GPU process due to errors during initialization
^C/home/pi/MagicMirror/node_modules/electron/dist/electron exited with signal SIGINT

**Comment by mdefrancesco** [comment]
ok so i removed the module which was giving me the problem

now i have another couple of issues

i cant remove the week numbers (CW 50) so the best i could do was remove the "CW " and make it invisible. i tried the code in custom css but it didnt change anything

.CX3 .cw {
  display: none;
}

i can change the background, i can change the broder color around today but i cant change any other colors.

i'm using 3 calendars and wanted them to have individual colors like i have in other calendar modules but it's not working, i have the colors established in the calendar module in the config.js and also tried adding the color values to this module in the config...but nothing

then i tried changing any entries that said "Chloe work" to red using the following the custom css and nothing

eventTransformer: (event) => {
  if (event.title.includes('Chloe work')) {
    event.title = `<span style="color: red !important;">${event.title}</span>`;
  }
  return event;
}

sorry to be a bother, but i really like the module and am trying to make it work

![image](https://github.com/MMRIZE/MMM-CalendarExt3/assets/123851794/4b97302a-4ef8-4523-9b36-472d334e0aca)

**Reply by mdefrancesco** [reply]
and where are you establishing color values like "crimson" and "dodgerblue"?

**Reply by mdefrancesco** [reply]
looking back at the calendar it seems like it is changing the background only for all day events. not sure if that helps

**Comment by eouia** [comment]
Sorry for late reply. It was holidays. :D

1. `CX3 .cw { display: none }` is a right, intended way to do that. For the presentation, to use CSS is my intentional design. 

2. each cell (the square of the day) has many values of properties as classes and attributes.
This is example;
```html
<div class="cell today thisMonth thisYear year_2023 month_12 date_26 weekday_2" data-date="1703545200000" data-events="0" data-has-events="false" id="1703545200000_1703606470290" data-popoverble="true"><div class="cellHeader"><div class="cw">52</div><div class="cellDate"><span class="dateParts month seq_0">Dec</span><span class="dateParts literal seq_1"> </span><span class="dateParts day seq_2">26</span></div></div><div class="cellBody"></div><div class="cellFooter"></div></div>
``` 
So you can select the cell of today like this;
```css
.CX3 .cell.today {
    border: 1px solid #fff;
    background-color: gray;
}
```
<img width="312" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/237b0028-0f7a-4cc1-90b4-cd506b84011a">

https://github.com/MMRIZE/MMM-CalendarExt3?tab=readme-ov-file#styling-with-css

3. 
> i'm using 3 calendars and wanted them to have individual colors like i have in other calendar modules but it's not working, i have the colors established in the calendar module in the config.js and also tried adding the color values to this module in the config...but nothing

It's weird. The color you assigned in your default calendar module(or whichever calendar event provider module), it should be applied in this module.
```js
{
	module: "calendar",
	header: "US Holidays",
	position: "top_left",
	config: {
		calendars: [
			{
				name: "US Holidays",
				symbol: "calendar-check",
				color: "green",
				url: "https://ics.calendarlabs.com/76/mm3137/US_Holidays.ics"
			}
		]
	}
},
```
<img width="297" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/b276460a-ebd0-4a6a-a674-0e74b23849cc">

4. Of course, `eventTransformer` could do it, I'm not sure why your approach didn't work. 
```js
{
	module: "MMM-CalendarExt3",
	position: "bottom_bar",
	title: "my calendars",
	config: {
		mode: "month",
		displayLegend: true,
		eventTransformer: (event) => {
			if (event.title.includes("Day")) event.color = "red"; // Originally it was "green"
				return event;
			}
                 }
	}
},
```
<img width="297" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/654203ad-3c79-47a0-beba-bceda13365be">


5. For holidays, I preassigned them as defaults colors;
```css
.CX3 .weekend {
  color: crimson;
  font-weight: bold;
}

.CX3 .weekend_2 {
  color: crimson;
}

.CX3 .weekend_1 {
  color: dodgerblue;
}
```
You can override them in your `custom.css`.

6. 
> looking back at the calendar it seems like it is changing the background only for all day events. not sure if that helps

By default, `not full day or multiday event`(single day event) doesn't have colour on the title. I hate the colourful text, so only `symbol` has the colour defined by config.
But if you want it, use this CSS orverriding.
```css
/* In your custom.css */
.CX3 .event .headline .title {
    color: var(--calendarColor);
}
``` 
<img width="165" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/714658cc-edd2-413e-88d4-8a696af5cec7">

**Comment by mdefrancesco** [comment]
happy holidays, glad you stepped away. thanks again for your attention to this

trying to pick these apart one by one

**i started with a "clean" config.js and completely removed the custom CSS

the symbols (which i had removed) are reflecting proper color and also are reflecting the event transformer
all day events have the colored background and "timed" events have the colored symbol.

arghhh!! i feel like its so close
![image](https://github.com/MMRIZE/MMM-CalendarExt3/assets/123851794/1713df34-5ba7-4305-9abc-43362d541ba3)


here is the "clean" config.js file

/* MagicMirror² Config Sample
 *
 * By Michael Teeuw https://michaelteeuw.nl
 * MIT Licensed.
 *
 * For more information on how you can configure this file
 * see https://docs.magicmirror.builders/configuration/introduction.html
 * and https://docs.magicmirror.builders/modules/configuration.html
 */
let config = {
	address: "localhost", 	// Address to listen on, can be:
							// - "localhost", "127.0.0.1", "::1" to listen on loopback interface
							// - another specific IPv4/6 to listen on a specific interface
							// - "0.0.0.0", "::" to listen on any interface
							// Default, when address config is left out or empty, is "localhost"
	port: 8080,
	basePath: "/", 	// The URL path where MagicMirror² is hosted. If you are using a Reverse proxy
					// you must set the sub path here. basePath must end with a /
	ipWhitelist: ["127.0.0.1", "::ffff:127.0.0.1", "::1"], 	// Set [] to allow all IP addresses
															// or add a specific IPv4 of 192.168.1.5 :
															// ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.1.5"],
															// or IPv4 range of 192.168.3.0 --> 192.168.3.15 use CIDR format :
															// ["127.0.0.1", "::ffff:127.0.0.1", "::1", "::ffff:192.168.3.0/28"],

	useHttps: false, 		// Support HTTPS or not, default "false" will use HTTP
	httpsPrivateKey: "", 	// HTTPS private key path, only require when useHttps is true
	httpsCertificate: "", 	// HTTPS Certificate path, only require when useHttps is true

	language: "en",
	locale: "en-US",
	logLevel: ["INFO", "LOG", "WARN", "ERROR"], // Add "DEBUG" for even more logging
	timeFormat: 12,
	units: "imperial",
	// serverOnly:  true/false/"local" ,
	// local for armv6l processors, default
	//   starts serveronly and then starts chrome browser
	// false, default for all NON-armv6l devices
	// true, force serveronly mode, because you want to.. no UI on this device

	modules: [
		{
			module: "alert",
		},
		{
			module: "updatenotification",
			position: "top_bar"
		},
		{
			module: "clock",
			position: "top_left"
		},
		{
			module: "calendar",
			header: "US Holidays",
			//position: "top_left",
			config: {
				   
    broadcastPastEvents: true,
    maximumEntries: 1000, // There was a change just before the release of 2.20.0 that may require setting this high to see all of your future events
    displaySymbol: false,
    wrapEvents: true,
    maxTitleLines: 20,
    colored: true,
				  calendars: [
      
            {
		symbol: "calendar",
		name: "us_holiday",
		url: "https://calendar.google.com/calendar/ical/en.usa%23holiday%40group.v.calendar.google.com/public/basic.ics",
		color: "#FFFF00",

			},
	    {
		//symbol: "fa-calendar-o",
		name: "ellen",
		url: "https://calendar.google.com/calendar/ical/xxxxxxxxxxxxxxxxxxxxxxxxxxx",
		color: "#90EE90",
			},

	    {	
		//symbol: "fa-birthday-cake",
		name: "mark",
		url: "https://calendar.google.com/calendar/icalxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
		color: "#6AC5FE",
	    },
      

    ]
			}
		},
		{
			module: "compliments",
			position: "lower_third"
		},
		
		
		//NEW CALENDAR EXRTENSION 3
{
		module: "MMM-CalendarExt3",	
		position: "top_bar",
		title: "calendarpalooza",
		config: {
				mode: "week",
				//weekIndex: 3, 
				weeksInView: 8,
				fontSize: 23,
				displayLegend: true,
				eventTransformer: (event) => {
						if (event.title.includes("Chloe work")) event.color = "red"; // Originally it was "green"
				return event;
			},
				
				//instanceId: "basicCalendar",
				//locale: 'de-DE',
				maxEventLines: 6,
				firstDayOfWeek: 1,
				calendars: [
          ],
				calendarSet: ['us_holiday', 'ellen', 'mark'],
  }
},
		
		
		
		
		
		{
			module: "weather",
			header: "Buffalo",
			position: "top_right",
			config: {
				weatherProvider: "openweathermap",
				type: "current",
				location: "New York",
				locationID: "5128581", //ID from http://bulk.openweathermap.org/sample/city.list.json.gz; unzip the gz file and find your city
				apiKey: "xxxxxxxxxxxxxxxxxxxxxxxxxx"
			}
		},

		{
			module: "newsfeed",
			position: "bottom_bar",
			config: {
				feeds: [
					{
						title: "New York Times",
						url: "https://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml"
					}
				],
				showSourceTitle: true,
				showPublishDate: true,
				broadcastNewsFeeds: true,
				broadcastNewsUpdates: true
			}
		},
	]
};

/*************** DO NOT EDIT THE LINE BELOW ***************/
if (typeof module !== "undefined") {module.exports = config;}

**Reply by eouia** [reply]
So... all your issues are solved now?

**Comment by mdefrancesco** [comment]
i wish ;-)

so actually none of them are solved

1.  added this to custom.css and did not work, still shows up
.CX3 .cw {
  display: none;
  //visibility: hidden;
  //color: blue;
}

2. i am able to make changes to the individual cell background colors, borders, etc. so thats not an issue

3. assiging the colors to the indivudal calendar entries is still my biggest issue. when i remove the custom css completely the symbols will pick up the correct color but there is no background color change and no text color change. this is literally the biggest thing that is driving me crazy. *****seems to work properly for all-day and multi-day events but individual events it does not work. i started playing around in the 
MMM-CalendarExt3.css but i didnt think that was a good idea.

4. when i add the event transformer to the config.js and remove the custom.css it works. ***i'll add my original custom.css at the end of this reply
5. my question was...something like dodgerblue...where in MagicMirror are those hex color values, i'm just curious. when i assign something "pink" where does it get those color values from?
6. so that works...but there would have to be a way to turn off all the full day and multi day events which will now have the text color match the background color :-)


i dont mean to be a pain in the butt, but thanks for all of your help with this

**Reply by mdefrancesco** [reply]
body {
	margin: 0;
	position: absolute;
 	transform: rotate(-90deg);
 	transform-origin: top left; 
	width: 100vh;
	height: 100vw;
	object-fit: cover;
	top: 100vh;
        visibility: visible;
 }


.clock .time {
 font-size: 100px;
}

.clock .date {
 font-size: 50px;
}

.MMM-WOTD header {

  font-size: 20px;
  color: white;
  }

eventTransformer: (ev) => {
  if (ev.title.search('Chloe work') > -1) ev.color = 'blue'
  return ev
},

  
.MMM-WOTD {
    width: 400px;
    margin-top: 100px;
    
}

.MMM-History .header {
	text-align: left;
}
.MMM-History .title {
	text-align: left;
}

.MMM-History {
    text-align: left;
}

.CX3 {
  /* you CAN modify these values; but SHOULD NOT to remove */
  --celllinecolor: #0d0d0d;
  
}

.CX3 .event .headline.useSymbol .symbol {
  display: none;
}

.CX3 .cw {
  display: none;
  visibility: hidden;
  color: blue;
}

.CX3 .cw::before {
  content: ''
}

.CX3 .weekend {
  color: gray;
  font-weight: bold;
}

.CX3 .cell.today {
  border: 1px solid #90EE90;
}

.CX3 .thisMonth {
  background-color: rgba(255, 0, 128, 0.1);
}

.MMM-Dad-Jokes {
    width: 400px;
    text-align: left;
    margin-top: 100px;
    
}
.MMM-Dad-Jokes header{
    text-align: left;
    font-size: 20px;
  color: white;
{

**Reply by eouia** [reply]
1. 
First, your `eventTransformer` function should not be in `custom.css`. It is javascript code, not css. It should be placed inside of your module config.
![image](https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/62286a8a-fd0e-4dbe-9294-c321d3d75f3c)

2. To remove CW, try this.
```css
/* custom.css */
.CX3 .cellContainer .cell:nth-child(1) .cw {
  display: none;
}
```
I forgot I had revealed CW on the first cell of the weekdays. I think this will work.

3. For the style of the event, you have to understand the default style first.
- full-day events or multi-day events would have a "coloured" background and white/black(depending on the calendar colour `--calendarColor`) text and symbol.
- single day "timed" event would have the color symbol and white(`--defaultColor`) text.

To decorate the event, there are two dynamic magic values, `--calendarColor` and `-oppositeColor`.
So, what you want is "colored text on single-day-timed-event",  use this CSS.
```css
/* custom.css */
.CX3 .event.singleday .headline .title {
  color: var(--calendarColor);
}
```
<img width="437" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/f7cba7a9-f944-4b65-bde3-b4987143bf1f">

And you don't have to use `eventTransformer` for this work.

4. Never edit the module's files. (.css or .js, ...). All you can modify would be `config.js` and `custom.css`. Otherwise, it will be the issue on updates.

5. `pink`, `dodgerblue` are pre-defined colour values by HTML/CSS itself. 

https://developer.mozilla.org/en-US/docs/Web/CSS/named-color

Of course, you can use HEX colour instead named colour value like `#FFC0CB` instead of `pink`.

>so that works...but there would have to be a way to turn off all the full day and multi day events which will now have the text color match the background color :-)

To prevent that kind of mistake, I prepared `--calendarColor` and `oppositeColor`.

By default;

```css
.CX3 .event.fullday,
.CX3 .event.multiday {
  background-color: var(--calendarColor); /* Color by config or eventTransformer */
  color: var(--oppositeColor); /* Auto calculated color by --calendarColor */
  ...
}
```

```css
.CX3 .event { /* singleday event */
  color: var(--defaultcolor);
  ...
}
```

So try my answer No.3.

**Comment by mdefrancesco** [comment]
holy crap! i think you fixed everything.
i just have one more question re: event transformer. 

yes, it works when i put in the config and not the custom css

how would i change it to instead of changing the color of the text i wanted to give it a background color (like the full and multi-day events) and give the text the opposite color. I've been trying to figure out the wording but nothing is working

**Reply by mdefrancesco** [reply]
i tried, but this doesnt work

{
		module: "MMM-CalendarExt3",	
		position: "top_bar",
		title: "calendarpalooza",
		config: {
				mode: "week",
				//weekIndex: 3, 
				weeksInView: 8,
				fontSize: 23,
				displayLegend: true,
				eventTransformer: (event) => {
      if (event.title.includes("Chloe work")) {
      event.backgroundColor = "red";
      event.textColor = "white";
    }
				return event;
			},

**Reply by eouia** [reply]
For all singleday events? Or for specific calendar? Or for specific events?(e,g. Only for the events have “birthday” title)

**Reply by mdefrancesco** [reply]
so in this case i am trying to assign the background color and text color adjustments to any events that have "Chloe work" in the title

**Reply by mdefrancesco** [reply]
and if possible to turn on the symbols during this event transform as well. (i turned them off everywhere esle in the custom css)

**Reply by eouia** [reply]
- Original 

<img width="220" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/9bfd39ac-fbae-4908-b7d6-8e0bbe2b6c7f">

---

- Modified.

```js
eventTransformer: (ev) => {
	if (ev.title.includes('Chloe work')) ev.fullDayEvent = true
	return ev
},
```
<img width="220" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/98e29d13-36a5-472e-b61b-9d73ce9076b3">

**Reply by mdefrancesco** [reply]
hmmmm...is there a way to keep the time ? seems like you changed it to mimic a full day event. the time is helpful because it lets us know what time she goes to work. would be awesome if in event transition you could include the end time as well

**Reply by eouia** [reply]
<img width="206" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/dbbd3ee2-9cdb-40ac-b578-f5b78a58b45f">

```js
/* config.js */
eventTransformer: (ev) => {
	if (ev.title.includes('Chloe work')) ev.class = "chloe"
	return ev
},
```


```css
/* custom.css*/
.CX3 .event.chloe {
  background-color: var(--calendarColor);
  color: var(--oppositeColor);
  border-radius: 4px;
}

.CX3 .event.singleday.chloe .headline .title {
  color: var(--oppositeColor);
}

.CX3 .event.singleday.chloe .headline.useSymbol .symbol {
  color: var(--oppositeColor);
}
```

**Reply by eouia** [reply]
And displaying end time is possible, but usually there is not enough space in the cell.(unless you are using 4k display)

**Reply by mdefrancesco** [reply]
so you turned chloe into a preset and assigned it where necessary...thats brilliant!!

last one, if i did want to include the end time, what would that look like?

**Reply by eouia** [reply]
```css
.CX3 .event.singleday.chloe .headline .time.endTime {
  display: inherit;
  order: 4;
  margin-left: auto;
}
```

W: 1080px;
<img width="103" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/a8bbb60d-6ebc-4aeb-b6da-08032122a76b">

W: 1920px;
<img width="149" alt="image" src="https://github.com/MMRIZE/MMM-CalendarExt3/assets/1720610/ac7062d9-1a47-40a5-8d9c-e4be3686ebec">

**Reply by mdefrancesco** [reply]
This is really amazing and definitely a learning experience. Thanks so much for all of your help and especially all of your patience. 

I "think" I have all of my issues addressed but i'll reach out again if any other questions pop up

You were an incredible help and I can't thank you enough

**Comment by mdefrancesco** [comment]
ok , a couple more quick ones.
is there a way to turn off symbols for specific calendars? i've tried what makes sense but its not working

and i had asked about abbreviating any times from 9:00 to just 9 in another thread and what you suggested didnt work

**Comment by eouia** [comment]
When you want to hide symbol of a specific calendar
```css
.CX3 .event.CUSTOMCLASS .headline.useSymbol .symbol { /* Replace "CUSTOMCLASS" to yours */
  display: none;
}
```

**Comment by mdefrancesco** [comment]
so in my config i have this

	    {
		symbol: "e",
		name: "ellen",
		displaySymbol: false,
		url: "https://calendar.google.com/calendar/ical/xxxxxxxxxxxxxxxxxxxbasic.ics",
		color: "#90EE90",
		class: 'ellen-calendar',
		
and then in my custom css i have this

.CX3 .event.ellen-calendar .headline.useSymbol .symbol { /* Replace "CUSTOMCLASS" to yours */
  display: none;
}

unfortunately it didnt do anything

**Comment by eouia** [comment]
That seems `calendar` module's config. You have confused these two modules' configs with each other. Those both do not share the same config. So you should place each config property in the proper position.

**Comment by mdefrancesco** [comment]
sorry, is this right? still doesnt seem to be working

		{
	module: "MMM-CalendarExt3",
	position: "bottom_bar",
	title: "",
	config: {
		mode: "month",
		eventTransformer: (event) => {
			if (event.fullDayEvent) return event
			let time = new Date(+event.startDate)
			let timeFormat = (time.getMinutes() === 0) ? { hour: 'numeric', hour12: true } : { hour: 'numeric', minute: '2-digit', hour12: true }
			let timeStr = time.toLocaleTimeString(['en-US'], timeFormat)
			event.title = `<span class="customTime">${timeStr}</span>` + event.title
			return event; },
			
			calendars: [
				    {
        	url: "https://calendar.google.com/calendar/ical/xxxxxxxxxxxxxxxxxx/basic.ics",
				class: 'ellen-calendar',
      },
				 
						
          ],
		},
	
},

**Reply by eouia** [reply]
This module doesn't have `calendars: []` config.

**Reply by mdefrancesco** [reply]
im sorry, then i dont understand where to put it

**Reply by KristjanESPERANTO** [reply]
MMM-CalendarExt3 does not have a `calendars` config option — that belongs to the default `calendar` module. You cannot set a `class` there.

To hide symbols for a specific calendar, use `eventTransformer` in the MMM-CalendarExt3 config to assign a custom class based on the calendar name, then target it in `custom.css`:

**config.js** (inside MMM-CalendarExt3 config):
```js
eventTransformer: (event) => {
  if (event.calendarName === "ellen") event.class = "ellen-calendar"
  return event
},
```

**css/custom.css**:
```css
.CX3 .event.ellen-calendar .headline.useSymbol .symbol {
  display: none;
}
```
